rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinTestUtils.java

```diff
@@ -125,4 +128,31 @@ public static byte[] flatKeysAsByteArray(List<BtcECKey> keys) {
 
         return flatPubKeys;
     }
+
+    public static void signTransactionInputFromMultiSigWithKeys(BtcTransaction transaction, int inputIndex, List<BtcECKey> keys) {
+        if (transaction.getWitness(inputIndex).getPushCount() == 0) {
+            signLegacyTransactionInputWithP2shMultiSigInputScript(transaction, inputIndex, keys);
+        }
+    }
+
+    private static void signLegacyTransactionInputWithP2shMultiSigInputScript(BtcTransaction transaction, int inputIndex, List<BtcECKey> keys) {
+        TransactionInput input = transaction.getInput(inputIndex);
+
+        Optional<Script> inputRedeemScriptOpt = BitcoinUtils.extractRedeemScriptFromInput(input);
+        Script inputRedeemScript = Optional.of(inputRedeemScriptOpt)
+            .get()
+            .orElseThrow(() -> new IllegalArgumentException("Cannot sign inputs that are not from a multisig"));
+
+        Sha256Hash sigHash = transaction.hashForSignature(inputIndex, inputRedeemScript, BtcTransaction.SigHash.ALL, false);
+        Script inputScriptSig = input.getScriptSig();
+        for (BtcECKey key : keys) {
+            BtcECKey.ECDSASignature sig = key.sign(sigHash);
+            TransactionSignature txSig = new TransactionSignature(sig, BtcTransaction.SigHash.ALL, false);
+            byte[] txSigEncoded = txSig.encodeToBitcoin();
+
+            Script outputScript = createP2SHOutputScript(inputRedeemScript);
+            inputScriptSig = outputScript.getScriptSigWithSignature(inputScriptSig, txSigEncoded, keys.indexOf(key));
```

Comment:
Not sure about this way of getting the index, shouldn't we be using `getSigInsertionIndex`?

Reply:
you are right! For the method to work irl when signing txs, we should get the sig as you say. Thanks!!